pkg/k8s/k8s_pod.go

```diff
@@ -417,18 +418,38 @@ func buildInitContainerVolumes(name string, volumes []*Volume) ([]v1.VolumeMount
 }
 
 // buildInitContainerCommand generates a command for an init container based on the given name and volumes.
-func buildInitContainerCommand(volumes []*Volume) ([]string, error) {
-	if len(volumes) == 0 {
-		return []string{}, nil // return empty slice if no volumes are specified
+func buildInitContainerCommand(volumes []*Volume, files []*File) ([]string, error) {
+	var commands = []string{"sh", "-c"}
+	dirsProcessed := make(map[string]bool)
+	baseCmd := "mkdir -p /knuu && "
+	cmds := []string{baseCmd}
+
+	for _, file := range files {
+		// get the directory of the file
+		folder := filepath.Dir(file.Dest)
+		if _, processed := dirsProcessed[folder]; !processed {
+			parentDirCmd := fmt.Sprintf("mkdir -p /knuu%s && chmod -R 777 /knuu/%s && ", folder, folder)
```

Comment:
Can you please double check if we really need that `/` before `%s` or not?
It seems like there is no `/` in `mkdir` and there is one in `chmod`
we can also use [path.Join](https://pkg.go.dev/path#Join)